imports/plugins/core/email/client/components/emailConfig.js

```diff
@@ -17,6 +20,21 @@ class EmailConfig extends Component {
     this.toggleSettings = this.toggleSettings.bind(this);
   }
 
+  componentWillMount() {
+    const { settings } = this.props;
+    const { service, host, port, user, password } = settings;
+    // if all settings exist, check if they work
+    if (service && host && port && user && password) {
+      Meteor.call("email/verifySettings", (error) => {
+        if (error) {
+          this.setState({ status: "error", error: error.reason });
+        }
+        this.setState({ status: "valid", error: null });
+      });
+    } else {
+      this.setState({ status: "error", error: null });
+    }
+  }
```

Comment:
This kind of logic should not be in a UI component.  Data retrieval and/or interaction with the server should always be done in the container component and then results passed into the UI component.  The UI component should only have markup and interactivity handling (`onClick`, etc).